spec-tests/src/test/java/org/candlepin/spec/InactiveConsumerCleanerJobTest.java

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright (c) 2009 - 2022 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package org.candlepin.spec;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.candlepin.ApiException;
+import org.candlepin.dto.api.v1.AsyncJobStatusDTO;
+import org.candlepin.dto.api.v1.ConsumerDTO;
+import org.candlepin.dto.api.v1.ConsumerTypeDTO;
+import org.candlepin.dto.api.v1.NestedOwnerDTO;
+import org.candlepin.dto.api.v1.OwnerDTO;
+import org.candlepin.spec.bootstrap.client.ConsumerClient;
+import org.candlepin.spec.bootstrap.client.ConsumerTypeClient;
+import org.candlepin.spec.bootstrap.client.JobsClient;
+import org.candlepin.spec.bootstrap.client.OwnerClient;
+import org.candlepin.spec.bootstrap.client.SpecTestFixture;
+import org.candlepin.spec.bootstrap.data.ConsumerDTOBuilder;
+import org.candlepin.spec.bootstrap.data.OwnerDTOBuilder;
+import org.candlepin.spec.bootstrap.data.Util;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.time.temporal.ChronoUnit;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class InactiveConsumerCleanerJobTest extends SpecTestFixture {
+
+    private static final String JOB_KEY = "InactiveConsumerCleanerJob";
+    private static final int DEFAULT_LAST_CHECKED_IN_RETENTION_IN_DAYS = 397;
+
+    private ConsumerClient consumerClient;
+    private OwnerClient ownerClient;
+    private JobsClient jobsClient;
+    private ConsumerTypeClient consumerTypeClient;
+
+    private OwnerDTO owner;
+    private ConsumerTypeDTO consumerType;
+
+    @BeforeEach
+    private void beforeEach() throws ApiException {
+        consumerClient = getClient(ConsumerClient.class);
+        ownerClient = getClient(OwnerClient.class);
+        jobsClient = getClient(JobsClient.class);
+        consumerTypeClient = getClient(ConsumerTypeClient.class);
+
+        if (owner == null) {
+            owner = ownerClient.createOwner(new OwnerDTOBuilder().build());
```

Comment:
Take a look at Owners class. It provides factory methods for randomized owner instances.

Reply:
Awesome! Used Owners.random()